tools/src/test/scala/scala/scalanative/IncCompilationTest.scala

```diff
@@ -77,7 +77,7 @@ class IncCompilationTest extends codegen.CodeGenSpec with Matchers {
       )
       val entry = "A"
       val changedTop = Set[String]("A", "A$")
-      val outDir = Files.createTempDirectory("native-test-out")
+      val outDir = Files.createTempDirectory("native-test-out1")
```

Comment:
The need to modify the output directory is for me a signal something very wrong was happening here. Yes, it can admit that perfectly each test should be executed in a different, clean environment, but in that case, it also perfectly shows the case of incremental editing and rebuilding the sources. We should never be forced to create new temp directory each time.

Reply:
This is just showing a crack in the caching approach but really we shouldn't be reusing build directories I don't think - the multiple main class works but that is a special case. However, if our build was super good it might clean as needed to keep all the artifacts up to date and in sync.